appserver/tests/payara-samples/test-domain-setup/src/test/java/fish/payara/samples/RestartDomain.java

```diff
@@ -51,12 +53,12 @@
 public class RestartDomain {
     private static final Logger logger = Logger.getLogger(RestartDomain.class.getName());
 
-    public static boolean NEEDS_RESTART;
-
     public static void restart() {
```

Comment:
Should be renamed to "restartIfRequiredByServer". Or even moved to ServerOperations.
Also server sometimes requires restart, but get-restart-required may return false, because the server doesn't know it. Then user can directly call ServerOperations.restartDomain, so ok, just the name of the RestartDomain class name and it's method pretends it does the same.

Reply:
The whole purpose of this is to *not* let users restart the domain